src/gui/transferlistsortmodel.cpp

```diff
@@ -186,15 +187,16 @@ bool TransferListSortModel::lessThan(const QModelIndex &left, const QModelIndex
 {
     Q_ASSERT(left.column() == right.column());
 
-    if (m_sortColumn != left.column())
+    if (m_lastSortColumn != left.column())
     {
-        m_subSortColumn = m_sortColumn;
-        m_sortColumn = left.column();
+        m_subSortColumn = m_lastSortColumn;
+        m_lastSortColumn = left.column();
     }
 
     const int result = compare(left, right);
-    if ((result == 0) && (m_subSortColumn != -1))
-        return compare(left.sibling(left.row(), m_subSortColumn), right.sibling(right.row(), m_subSortColumn)) < 0;
+    const int subSortColumn = m_subSortColumn.get();
+    if ((result == 0) && (subSortColumn >= 0 && subSortColumn < columnCount()))
```

Comment:
There are 2 ways to handle untrusted inputs: 1. use `ProxyFunc` from `CachedSettingValue` constructor. 2. ignore it, it's users' own fault breaking it.
I would choose the latter way.
And add the required parentheses if you chose the first.

Reply:
this can serve as a check when the transfer list column changes in the future.

Reply:
`ProxyFunc` doesn't apply here since we don't have source model available at the point of initialization.

Reply:
>this can serve as a check when the transfer list column changes in the future.

I don't think the check is correct. For example: at some time, only 2 columns are enabled and `subSortColumn == TR_AVAILABILITY`.

Reply:
> at some time, only 2 columns are enabled and subSortColumn == TR_AVAILABILITY.

afaik for models, all columns are enabled at all time, views hides/disables columns at their own level. Also, I don't want to change sub-sort column based on the visibility of the column, that's too much.

Reply:
>afaik for models, all columns are enabled at all time, views hides/disables columns at their own level. 

Right, I didn't noticed this isn't the view.

>ProxyFunc doesn't apply here since we don't have source model available at the point of initialization.

Seems you can use `Column::NB_COLUMNS` within `ProxyFunc`.

>Also, I don't want to change sub-sort column based on the visibility of the column, that's too much.

Maybe instead of defaulting subsort column to `-1`, you can default it to `TR_NAME`?